hil/config.py

```diff
@@ -8,13 +8,94 @@
 import ConfigParser
 import logging.handlers
 import importlib
+from schema import Schema, Optional
 import os
 import sys
+from urlparse import urlparse
 
 cfg = ConfigParser.RawConfigParser()
 cfg.optionxform = str
 
 
+def string_is_bool(option):
+    """Check if a string matches ConfigParser's definition of a bool"""
+    return option.lower() in ['true', 'yes', 'on', '1',
+                              'false', 'no', 'off', '0']
+
+
+def string_is_web_url(option):
+    """Check if a string is a valid web URL"""
+    url = urlparse(option)
+    return not (url.scheme == '' or url.netloc == '')
+
+
+def string_is_db_uri(option):
+    """Check if a string is a valid DB URI"""
+    url = urlparse(option)
+    if url.scheme not in ('postgresql', 'sqlite'):
+        return False
+    if url.path == '':
```

Comment:
why not `or` this with the previous statement?
```python
if url.scheme not in ('postgresql', 'sqlite') or url.path == '':
```
or maybe directly return it.
```python
def string_is_db_uri(option):
    return url.scheme in ('postgresql', 'sqlite') and url.path != ''
```

Additionally, `'sqlite://'` is valid and gives an [in-memory database](http://docs.sqlalchemy.org/en/latest/dialects/sqlite.html#connect-strings), so `url.path` will be `''`.  But I don't mind making it mandatory for developers to specify `:memory:` in their tests (and that only makes sense, if your config checker is called before our tests that use in memory db).

Also, like in the case of logging directory where we want to make sure that the directory exists, we might wanna check that the database specified in the config file is actually reachable. This could just be an additional check if folks agree.
A simple function like this could check it:
```python
def is_db_responsive(option):
    try:
        db = create_engine(option)
        db.connect()
        db.dispose()
        return True
    except:
        return False
```

Reply:
Might be worth doing a check on start-up, but I think it should be separate from validating the config file.

Reply:
@naved001 I agree that we should check for an existing directory, but I also agree with @zenhack that we should do it outside of the config validation.  The config validation should really just be checking config strings.  This would fix #976.

I also see what you mean about the in-mem DBs too. I'm just going to check the scheme then, no need to limit users that much.

Reply:
I can just do a `os.path.isdir("/home/el")`